File: sdk-python/tests/iwfcompat/internal_channel_test.py
# Portions of this file are derived from indeedeng/iwf-java-sdk.
# Those portions are licensed under the Apache License, Version 2.0.
# See LICENSES/Apache-2.0.txt and LEGACY_NOTICES.md.
#
# Modifications Copyright (c) 2026 Super Durable, Inc.
#
# Modifications are licensed under the Super Durable Source License 1.0.
# Third-Party Materials remain under the Apache License, Version 2.0.
# See LICENSE and LEGACY_NOTICES.md.

from dex import Client

from . import iwf_flows


def compile_basic_internal_channel(client: Client) -> None:
    client.start_flow(iwf_flows.BASIC_INTERNAL, "basic-internal", 1)
    output: int = client.wait_for_flow("basic-internal", int)
    del output


def compile_waiting_internal_channel(client: Client) -> None:
    flow = iwf_flows.WAITING_INTERNAL
    client.start_flow(flow, "waiting-internal", 1)
    client.publish("waiting-internal", flow.channel, 2, 3)
    output: int = client.wait_for_flow("waiting-internal", int)
    del output

File: sdk-typescript/test/integ/internal_channel_test.ts
// Portions of this file are derived from indeedeng/iwf-java-sdk.
// Those portions are licensed under the Apache License, Version 2.0.
// See LICENSES/Apache-2.0.txt and LEGACY_NOTICES.md.
//
// Modifications Copyright (c) 2026 Super Durable, Inc.
//
// Modifications are licensed under the Super Durable Source License 1.0.
// Third-Party Materials remain under the Apache License, Version 2.0.
// See LICENSE and LEGACY_NOTICES.md.

import { doubleCodec, type Client } from "../../src/index.js";

import * as flows from "./iwf_flows.js";

export async function compileBasicInternalChannel(client: Client): Promise<void> {
  await client.startFlow(flows.BASIC_INTERNAL, "basic-internal", 1);
  const output: number = await client.waitForFlow("basic-internal", doubleCodec);
  void output;
}

export async function compileWaitingInternalChannel(client: Client): Promise<void> {
  const flow = flows.WAITING_INTERNAL;
  await client.startFlow(flow, "waiting-internal", 1);
  await client.publish("waiting-internal", flow.channel, 2, 3);
  const output: number = await client.waitForFlow("waiting-internal", doubleCodec);
  void output;
}
